testing/run_tests.py

```diff
@@ -605,25 +605,53 @@ def RunObjcTests(ios_variant='ios_debug_sim_unopt', test_filter=None):
   ios_out_dir = os.path.join(out_dir, ios_variant)
   EnsureIosTestsAreBuilt(ios_out_dir)
 
-  ios_unit_test_dir = os.path.join(
-      buildroot_dir, 'flutter', 'testing', 'ios', 'IosUnitTests'
-  )
+  new_simulator_name = 'IosUnitTestsSimulator'
 
-  # Avoid using xcpretty unless the following can be addressed:
-  # - Make sure all relevant failure output is printed on a failure.
-  # - Make sure that a failing exit code is set for CI.
-  # See https://github.com/flutter/flutter/issues/63742
-  command = [
-      'xcodebuild '
-      '-sdk iphonesimulator '
-      '-scheme IosUnitTests '
-      "-destination platform='iOS Simulator,name=iPhone 11' "
-      'test '
-      'FLUTTER_ENGINE=' + ios_variant
+  # Delete simulators with this name in case any were leaked
+  # from another test run.
+  DeleteSimulator(new_simulator_name)
+
+  create_simulator = [
+      'xcrun '
+      'simctl '
+      'create '
+      '%s com.apple.CoreSimulator.SimDeviceType.iPhone-11' % new_simulator_name
+  ]
+  RunCmd(create_simulator, shell=True)
+
+  try:
+    ios_unit_test_dir = os.path.join(
+        buildroot_dir, 'flutter', 'testing', 'ios', 'IosUnitTests'
+    )
+    # Avoid using xcpretty unless the following can be addressed:
+    # - Make sure all relevant failure output is printed on a failure.
+    # - Make sure that a failing exit code is set for CI.
+    # See https://github.com/flutter/flutter/issues/63742
+    test_command = [
+        'xcodebuild '
+        '-sdk iphonesimulator '
+        '-scheme IosUnitTests '
+        "-destination platform='iOS Simulator,name=iPhone 11' "
+        'test '
+        'FLUTTER_ENGINE=' + ios_variant
+    ]
+    if test_filter != None:
+      test_command[0] = test_command[0] + " -only-testing:%s" % test_filter
+    RunCmd(test_command, cwd=ios_unit_test_dir, shell=True)
+  finally:
+    DeleteSimulator(new_simulator_name)
```

Comment:
This isn't technically needed to fix the issue and it will increase run times for tests no?

Reply:
You're right, this is technically not needed since any leftover simulators are deleted before the next run of this script (which happens during flake detection), but I'd rather not leak the simulator to other scripts running simulator tests (scenario?)
Fortunately it's very fast, as mentioned above:
```
xcrun simctl delete test1 0.08s user 0.08s system 86% cpu 0.180 total
```